### PR TITLE
67 additional command annotations

### LINKED ISSUE #67: Additional command annotations
- [ ] @Description
- [ ] @LongDescription
- [ ] @Usage

### PR DESCRIPTION
**Pull request by CyberRei**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation)
- [ ] 🐞 Bug fix (fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)

### 📚 Description

Added additional command annotations
Resolves #67 

Could check docs [here](https://github.com/CyberRei/cocoa-beans/blob/67-additional-command-annotations/Writerside/topics/Command-Information.md)

### 🧪 How Has This Been Tested?
CommandInfoForTest what could I say

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: Writerside/topics/Command-Information.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+**Table of content:**
+- [Introduction](#introduction)
+- [Annotations](#annotations)
+- [Usage](#usage-of-annotations)
+- [Managing multiple classes](#managing-multiple-descriptions-usages-and-long-descriptions)
+- [Relevant links](#relevant-links)
+
+## Introduction
+
+In Java applications, managing and documenting commands effectively is crucial for both developers and users. The `CommandInfo` class provides a set of powerful annotations designed to enhance the clarity and usability of command-line interfaces within Java applications. These annotations facilitate comprehensive documentation and streamline the process of defining commands.
```
Comment (liorsl) [inline Writerside/topics/Command-Information.md:12]: nice chatgpt

File: Writerside/topics/Command-Information.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    // etc...
+}
+```
+
+If we want to get `CommandInfo` for the class itself we should call
+`commandManger.getCommandInfo(commandName);`
+
+## Managing Multiple Descriptions, Usages, and Long Descriptions
+particularly when dealing with command-line interfaces, there are scenarios where multiple classes might define commands with the same name, or where command classes are extended. The `CommandInfo` class allows for multiple `@Description` annotations to accommodate these situations, ensuring comprehensive and accurate documentation.
```
Comment (liorsl) [inline Writerside/topics/Command-Information.md:99]: sentences start with upper case

File: commands/src/main/java/net/apartium/cocoabeans/commands/CommandInfo.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    /* package-private */ void addLongDescription(final LongDescription longDescription, boolean first) {
+        if (first) {
+            longDescriptions.add(0, longDescription);
+            return;
+        }
+
+        longDescriptions.add(longDescription);
+    }
+
+    /* package-private */ void serialize(Annotation[] annotations, boolean first) {
```
Comment (liorsl) [inline commands/src/main/java/net/apartium/cocoabeans/commands/CommandInfo.java:115]: rename to fromAnnotations

File: commands/src/main/java/net/apartium/cocoabeans/commands/CommandManager.java [hunk 4]
```diff
@@ -180,6 +180,16 @@ private boolean invokeException(HandleExceptionVariant handleExceptionVariant, C
     }
 
     private boolean invoke(CommandContext context, Sender sender, RegisteredCommandVariant registeredCommandVariant) {
+        context = new CommandContext(
```
Comment (liorsl) [inline commands/src/main/java/net/apartium/cocoabeans/commands/CommandManager.java:183]: umm what is that

**Review by liorsl — CHANGES_REQUESTED** [review]

File: Writerside/topics/Command-Information.md (hunk 1 above)
Comment (ikfir) [reply Writerside/topics/Command-Information.md:12]: Ty

File: Writerside/topics/Command-Information.md (hunk 2 above)
Comment (ikfir) [reply Writerside/topics/Command-Information.md:99]: Opss

File: commands/src/main/java/net/apartium/cocoabeans/commands/CommandManager.java (hunk 4 above)
Comment (ikfir) [reply commands/src/main/java/net/apartium/cocoabeans/commands/CommandManager.java:183]: Multiple variant each have its own command info

**Review by liorsl — APPROVED** [review]

**Pull request merged**